frameworks/react/__tests__/AdvancedPicture.test.tsx

```diff
@@ -98,10 +101,183 @@ describe('AdvancedPicture', () => {
     setTimeout(() => {
       expect(component.html()).toContain(
         '<picture>' +
+        '<img src="https://res.cloudinary.com/demo/image/upload/sample"><source media="(max-width: 500px)" sizes="80vw" ' +
+        'srcset="https://res.cloudinary.com/demo/image/upload/c_crop,w_500/dog"></picture>');
+      done();
+    }, 0);// one tick
+  });
+
+  it('autoOptimalBreakpoints: should generate default srcset when min-width and max-width not provided',
+    function (done) {
+      const component = mount(
+        <AdvancedPicture
+          autoOptimalBreakpoints
+          cldImg={defualtImage}
+          sources={[
+            {
+              image: sepiaImage
+            }
+          ]}
+        />);
+      setTimeout(() => {
+        expect(component.html()).toContain('<picture>' +
+        '<img src="https://res.cloudinary.com/demo/image/upload/sample"><source sizes="100vw" ' +
+        'srcset="https://res.cloudinary.com/demo/image/upload/e_sepia/c_scale,w_828/sample 828w,' +
+        'https://res.cloudinary.com/demo/image/upload/e_sepia/c_scale,w_1366/sample 1366w,' +
+        'https://res.cloudinary.com/demo/image/upload/e_sepia/c_scale,w_1536/sample 1536w,' +
+        'https://res.cloudinary.com/demo/image/upload/e_sepia/c_scale,w_1920/sample 1920w,' +
+        'https://res.cloudinary.com/demo/image/upload/e_sepia/c_scale,w_3840/sample 3840w">' +
+        '</picture>'
+        );
+        done();
+      }, 0);// one tick
+    });
+
+  it('autoOptimalBreakpoints: should generate single src when min-width > max-width', function (done) {
+    const component = mount(
+      <AdvancedPicture
+        cldImg={defualtImage} sources={[
+          {
+            maxWidth: 500,
+            minWidth: 700,
+            image: smallImage
+          }
+        ]}
+      />);
+    setTimeout(() => {
+      expect(component.html()).toContain(
+        '<picture><img src="https://res.cloudinary.com/demo/image/upload/sample">' +
+        '<source media="(min-width: 700px) and (max-width: 500px)" sizes="100vw" ' +
+        'srcset="https://res.cloudinary.com/demo/image/upload/c_crop,w_500/dog"></picture>'
+      );
+      done();
+    }, 0);// one tick
+  });
+
+  it('autoOptimalBreakpoints: should produce single src when min-width = max-width', function (done) {
+    const component = mount(
+      <AdvancedPicture
+        autoOptimalBreakpoints
+        cldImg={defualtImage}
+        sources={[
+          {
+            minWidth: 2000,
+            maxWidth: 2000,
+            image: sepiaImage
+          }
+        ]}
+      />);
+    setTimeout(() => {
+      expect(component.html()).toContain('<picture>' +
         '<img src="https://res.cloudinary.com/demo/image/upload/sample">' +
-        '<source srcset="https://res.cloudinary.com/demo/image/upload/c_crop,w_500/dog" sizes="80vw"' +
-        ' media="(max-width: 500px)">' +
-        '</picture>');
+        '<source media="(min-width: 2000px) and (max-width: 2000px)" sizes="100vw" ' +
```

Comment:
This means it will **only** work for devices with exact 2000px, I guess the test case is fine, but it's very unlikely to be used by anyone.

Reply:
@patrick-tolosa do you think we should remove this condition? I mean the user has to provide an exact minWidth and maxWidth for this case to happen so assuming this it makes sense to leave it as is..

Reply:
discussed and decided to leave as is